playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -56,17 +56,48 @@
         {{ avail_disk.stdout }} Kb available.
     when: (embedded_etcd | bool) and (etcd_disk_usage.stdout|int > avail_disk.stdout|int)
 
-  - name: Install etcd (for etcdctl)
-    action: "{{ ansible_pkg_mgr }} name=etcd state=latest"
-    when: not openshift.common.is_atomic | bool
+  # etcd3 in RHEL 7.3.0 does not Obsolstes: etcd, however etcd3-3.0.12 or later will
+  # so until that ships we need to carefully install etcd 3.0 to ensure we get fixes
+  # for backing up embedded etcd in 2.3
+  # see https://github.com/coreos/etcd/pull/4952 and https://bugzilla.redhat.com/show_bug.cgi?id=1382634
+  - name: Check for etcd package
+    rpm_q:
+      name: etcd
+      state: present
+    failed_when: false
+    register: etcd_rpmq
```

Comment:
This should probably be conditionalized as should the etcd2_installed set_fact below. For safety and readability I almost think we should push all of these embedded specific statements out to something like "embedded_etcd_backup_hacks.yml" and slap a conditional on it. This would help readibility IMO and might actually save us from some bugs in the future.